cpp/src/arrow/scalar_test.cc

```diff
@@ -1984,4 +1985,56 @@ TEST_F(TestExtensionScalar, ValidateErrors) {
   AssertValidationFails(scalar);
 }
 
+template <typename T>
```

Comment:
This is to save my local work temporarily. No worry, I'll still simplify the case as suggested and get rid of the typed test.